framework/src/systems/AuxiliarySystem.C

```diff
@@ -535,40 +667,61 @@ AuxiliarySystem::computeElementalVars(ExecFlagType type)
   }
 }
 
+template <typename AuxKernelType>
 void
-AuxiliarySystem::augmentSparsity(SparsityPattern::Graph & /*sparsity*/,
-                                 std::vector<dof_id_type> & /*n_nz*/,
-                                 std::vector<dof_id_type> & /*n_oz*/)
+AuxiliarySystem::computeNodalVarsHelper(
+    const MooseObjectWarehouse<AuxKernelType> & warehouse,
+    const std::vector<std::map<std::string, MooseVariableFEBase *>> & vars,
+    const PerfID & timer)
 {
-}
-
-Order
-AuxiliarySystem::getMinQuadratureOrder()
-{
-  Order order = CONSTANT;
-  std::vector<MooseVariableFEBase *> vars = _vars[0].fieldVariables();
-  for (const auto & var : vars)
+  if (warehouse.hasActiveBlockObjects())
   {
-    if (!var->isNodal()) // nodal aux variables do not need quadrature
+    TIME_SECTION(timer);
+
+    // Block Nodal AuxKernels
+    PARALLEL_TRY
     {
-      FEType fe_type = var->feType();
-      if (fe_type.default_quadrature_order() > order)
-        order = fe_type.default_quadrature_order();
+      ConstNodeRange & range = *_mesh.getLocalNodeRange();
+      ComputeNodalAuxVarsThread<AuxKernelType> navt(_fe_problem, warehouse, vars);
+      Threads::parallel_reduce(range, navt);
+
+      solution().close();
+      _sys.update();
     }
+    PARALLEL_CATCH;
   }
 
-  return order;
-}
+  if (warehouse.hasActiveBoundaryObjects())
+  {
+    TIME_SECTION(timer);
 
-bool
-AuxiliarySystem::needMaterialOnSide(BoundaryID bnd_id)
-{
-  return _elemental_aux_storage.hasActiveBoundaryObjects(bnd_id);
-}
+    // Boundary Nodal AuxKernels
+    PARALLEL_TRY
+    {
+      ConstBndNodeRange & bnd_nodes = *_mesh.getBoundaryNodeRange();
+      ComputeNodalAuxBcsThread<AuxKernelType> nabt(_fe_problem, warehouse, vars);
+      Threads::parallel_reduce(bnd_nodes, nabt);
 
-void
-AuxiliarySystem::setPreviousNewtonSolution()
-{
-  // Evaluate aux variables to get the solution vector
-  compute(EXEC_LINEAR);
+      solution().close();
```

Comment:
I just followed the design in place, didn't give it any thought.

Reply:
It should be possible to combine the vector and non-vector versions, but this would require major changes to the logic in the compute threads. I feel that this is beyond the scope of getting the core functionality in place. But, if you feel strong about it I can take a stab at it.